File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            int num2 = target - nums[i];
            if (map.containsKey(num2) && map.get(num2) != i) {
                return new int[]{i, map.get(num2)};
            }
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = [0] * len(nums)
        for index1, num1 in enumerate(nums):
            curr = 0
            for index2, num2 in enumerate(nums):
                if index1 != index2:
                    output[index1] += num2
        return output

"""
- Given an integer array nunms, return an array output, where output[i] is
the product of all the elements of nums except nums[i].
- Target time complexity is O(n). Space too.
- Challenge: Don't use division operation.
- Do your fist shot, then do follow-up. It's a follow-up, not an entirely
different question.
"""